File: EJERCICIO_1/EJERCICIOJAVA/arl.java
package EJERCICIOJAVA;

public class arl extends discount {
    private int risk;

    public arl(salaryBass salary, int risk) {
        super(salary, obtenerPorcentajePorRiesgo(risk));
        this.risk = risk;
    }

    private static double obtenerPorcentajePorRiesgo(int risk) {
        switch (risk) {
            case 1:
                return 0.0025;
            case 2:
                return 0.0050;
            case 3:
                return 0.0075;
            default:
                throw new IllegalArgumentException("Nivel de riesgo inválido");
        }
    }

    @Override
    public double calculateDiscount() {
        return salary.calculateSalary() * percentage;
    }

    public int getRisk() {
        return risk;
    }
}

File: EJERCICIO_1/EJERCICIOSPHP/arl.php
<?php

class Arl extends Discount {
    private $risk;

    public function __construct(SalaryBass $salary, $risk) {
        $this->risk = $risk;
        switch ($risk) {
            case 1:
                $percentage = 0.025;
                break;
            case 2:
                $percentage = 0.05;
                break;
            case 3:
                $percentage = 0.075;
                break;
            default:
                throw new Exception("Invalid risk level");
        }
        parent::__construct($salary, $percentage);
    }

    public function calculateDiscount(): float {
        return $this->salaryBass->calculateSalary() * $this->percentage;
    }

    public function getRisk(): int {
        return $this->risk;
    }
}
?>